core/src/main/java/org/geysermc/geyser/translator/collision/GlassPaneAndIronBarsCollision.java

```diff
@@ -0,0 +1,106 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.translator.collision;
+
+import lombok.EqualsAndHashCode;
+import org.geysermc.geyser.level.physics.BoundingBox;
+import org.geysermc.geyser.session.GeyserSession;
+
+@EqualsAndHashCode(callSuper = true)
+@CollisionRemapper(regex = "glass_pane$|iron_bars$", usesParams = true, passDefaultBoxes = true)
+public class GlassPaneAndIronBarsCollision extends BlockCollision {
+    /**
+     * 1 = north
+     * 2 = east
+     * 3 = south
+     * 4 = west
+     * 5 = north, east
+     * 6 = east, south
+     * 7 = south, west
+     * 8 = west, north
+     */
+    private int facing;
+
+    public GlassPaneAndIronBarsCollision(String params, BoundingBox[] defaultBoxes) {
+        super(defaultBoxes);
+        //east=true,north=true,south=true,west=true
+        if (params.contains("north=true") && params.contains("east=false") && params.contains("south=false") && params.contains("west=false")) {
+            facing = 1;
+        } else if (params.contains("north=true") && params.contains("east=true") && params.contains("south=false") && params.contains("west=false")) {
+            facing = 5;
+        } else if (params.contains("east=true") && params.contains("north=false") && params.contains("south=false") && params.contains("west=false")) {
+            facing = 2;
+        } else if (params.contains("east=true") && params.contains("north=false") && params.contains("south=true") && params.contains("west=false")) {
+            facing = 6;
+        } else if (params.contains("south=true") && params.contains("north=false") && params.contains("east=false") && params.contains("west=false")) {
+            facing = 3;
+        } else if (params.contains("south=true") && params.contains("north=false") && params.contains("east=false") && params.contains("west=true")) {
+            facing = 7;
+        } else if (params.contains("west=true") && params.contains("north=false") && params.contains("east=false") && params.contains("south=false")) {
+            facing = 4;
+        } else if (params.contains("west=true") && params.contains("north=true") && params.contains("east=false") && params.contains("south=false")) {
+            facing = 8;
+        }
```

Comment:
Looks like 2 `contains` calls can be removed from each if-expression

We aren't this strict/verbose in Door/Trapdoor collision

Reply:
Sorry, I was delayed because of some things!
Indeed, just as you said. This part seems like it could reduce some of the judgments, perhaps like this:
if (params.contains("north=true") && params.contains("east=true")) {
    facing = 5;
} else if (params.contains("east=true") && params.contains("south=true")) {
    facing = 6;
} else if (params.contains("south=true") && params.contains("west=true")) {
    facing = 7;
} else if (params.contains("west=true") && params.contains("north=true")) {
    facing = 8;
} else if (params.contains("north=true")) {
    facing = 1;
} else if (params.contains("east=true")) {
    facing = 2;
} else if (params.contains("south=true")) {
    facing = 3;
} else if (params.contains("west=true")) {
    facing = 4;
}


Or maybe using this method?
if (params.contains("north=true")) {
    facing |= 1; // 0001
}
if (params.contains("east=true")) {
    facing |= 2; // 0010
}
if (params.contains("south=true")) {
    facing |= 4; // 0100
}
if (params.contains("west=true")) {
    facing |= 8; // 1000
}
//North-East = 0011